src/common/modules/NetworkTools.js

```diff
@@ -57,8 +58,24 @@ export function fetch(input, init = {}, ...args) {
         }
     }
 
+    // add timeout value if needed
+    // from https://stackoverflow.com/a/50101022/5008962
+    const abortController = new AbortController();
+    let abortTimeoutId;
+
+    if (!init.signal) {
+        abortTimeoutId = setTimeout(() => abortController.abort(), 5000);
```

Comment:
I decided [against `AbortSignal.timeout`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout) as that is not widely implemented (only Firefox) and given https://github.com/rugk/mastodon-simplified-federation/issues/55 etc., this does not need to be adjusted then. It's also only a minor inconvenience in the code, so that's okay IMHO.

Though thinking about it, a note may be useful :D
```suggestion
        // replace with AbortSignal.timeout() if that is browser cross-compatible
        abortTimeoutId = setTimeout(() => abortController.abort(), 5000);
```